spec/lib/metasploit/framework/credential_collection_spec.rb

```diff
@@ -120,6 +121,37 @@
       end
     end
 
+    context "when given a pass_file and user_file and password spray" do
+      let(:password) { nil }
+      let(:username) { nil }
+      let(:password_spray) { true }
+      let(:user_file) do
+        filename = "user_file"
+        stub_file = StringIO.new("user1\nuser2\nuser3\n")
+        allow(File).to receive(:open).with(filename,/^r/).and_return stub_file
+
+        filename
+      end
+      let(:pass_file) do
+        filename = "pass_file"
+        stub_file = StringIO.new("password1\npassword2\n")
+        allow(File).to receive(:open).with(filename,/^r/).and_return stub_file
```

Comment:
Hi @nrathaus, thanks for the pull request. I have looked into the failing tests locally and I believe this is what is needed.

```suggestion
        allow(File).to receive(:open).with(filename,/^r/).and_yield stub_file
```

The previous implementation was returning the `pass_file` here:
https://github.com/rapid7/metasploit-framework/blob/13a79ab536b560f7aeb749afcb365f878794f170/lib/metasploit/framework/credential_collection.rb#L100

Then the credentials would be yielded within the loop:
https://github.com/rapid7/metasploit-framework/blob/13a79ab536b560f7aeb749afcb365f878794f170/lib/metasploit/framework/credential_collection.rb#L112

Where as now with your changes we will be doing the inverse of that logic. We return the `user_file` then yield the credentials when looping through the `pass_file`. Hopefully I explained that ok 😄